iati/fetch/__init__.py

```diff
@@ -0,0 +1,84 @@
+"""A module containing functionality for iati.fetch."""
+import iati.core.data
+import json
+import requests
+
+REGISTRY_API_METADATA_BY_DATASET_ID = "https://iatiregistry.org/api/3/action/package_show?id={0}"
+
+
+class Dataset(object):
+    """Container class to fetch a dataset."""
+
+    def __init__(self, dataset_id):
+        """Fetch a dataset by registry dataset ID.
+
+        Args:
+            dataset_id (str): The Registry UUID for the dataset.
+        """
+        self.dataset_id = dataset_id
+        self.dataset = None
+
+    def set_dataset(self):
+        """Set dataset from the dataset ID."""
+        registry_metadata = get_metadata(self.dataset_id)
+        dataset_url = registry_metadata['result']['resources'][0]['url']
+        dataset = get_dataset(dataset_url)
+        self.dataset = iati.core.data.Dataset(dataset)
+
+
+def get_metadata(dataset_id):
+    """Get registry metadata for given dataset ID.
```

Comment:
Need to determine preferred capitalisation of terms such as `Dataset`, `Schema`, `Codelist`, etc that are used within `iati.*`